syntax="proto3";

import "google/protobuf/timestamp.proto";

service ObjectService {
  rpc streaming_object(ObjectRequest) returns (stream StreamingData) {}
}

message ObjectRequest {
}

message StreamingData {
   repeated ObjectData object_data = 1;
   google.protobuf.Timestamp timestamp = 2;
}

message ObjectData {
  int32 object_id = 1; 
  ObjectType type = 2; 
  Coordinate coordinate = 3; //HD map 이미지 좌표내 Object 위치
}

message Coordinate {
  double x = 1;
  double y = 2;
}

enum ObjectType {
  Car = 0;
  Bus = 1;
  Truck = 2;
  AutonomousVehicle = 3;
  Bicycle = 4;
  Motorcycle = 5;
  PersonalMobility= 6;
  Pedestrian = 7;
}